xla/service/cpu/cpu_compiler.cc

```diff
@@ -711,12 +711,17 @@ Status CpuCompiler::RunHloPassesThroughLayoutAssn(
   // Convert BF16 and F8 operations to F32 and F16 respectively so that the CPU
   // backend can support BF16/F8 operations without directly implementing a
   // BF16/F8 lowering for most ops.
+  FloatSupport bf16_support(BF16);
+  CpuFloatSupport onednn_bf16_support(BF16);
```

Comment:
`CpuFloatSupport` is only defined when `INTEL_MKL` and `ENABLE_ONEDNN_V3` are defined. It must always be under the `#if defined(INTEL_MKL) && defined(ENABLE_ONEDNN_V3)` guard.

See an example error message in the [ARM CI](https://source.cloud.google.com/results/invocations/55cc0324-7842-4614-950b-2cddf4073439/log) failure:
```
xla/service/cpu/cpu_compiler.cc:715:3: error: unknown type name 'CpuFloatSupport'; did you mean 'FloatSupport'?
  715 |   CpuFloatSupport onednn_bf16_support(BF16);
      |   ^~~~~~~~~~~~~~~
      |   FloatSupport
./xla/service/float_support.h:28:7: note: 'FloatSupport' declared here
   28 | class FloatSupport {
      |       ^
1 warning and 1 error generated.
```

Reply:
Thanks for this info. Updated with moving the onednn_bf16_support under the guard.